app/Models/Foundation/Summit/Factories/SummitAttendeeNoteFactory.php

```diff
@@ -0,0 +1,78 @@
+<?php namespace models\summit\factories;
+/**
+ * Copyright 2023 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use Illuminate\Support\Facades\Log;
+use LaravelDoctrine\ORM\Facades\EntityManager;
+use models\exceptions\EntityNotFoundException;
+use models\main\Company;
+use models\main\Member;
+use models\summit\Summit;
+use models\summit\SummitAttendee;
+use models\summit\SummitAttendeeNote;
+
+/**
+ * Class SummitAttendeeNoteFactory
+ * @package models\summit\factories
+ */
+final class SummitAttendeeNoteFactory
+{
+    /**
+     * @param SummitAttendee $attendee
+     * @param array $payload
+     * @return SummitAttendeeNote
+     * @throws EntityNotFoundException
+     */
+    public static function build(SummitAttendee $attendee, array $payload)
+    {
+        return self::populate($attendee, new SummitAttendeeNote($payload['content'], $attendee), $payload);
+    }
+
+    /**
+     * @param SummitAttendee $attendee
+     * @param SummitAttendeeNote $attendee_note
+     * @param array $payload
+     * @return SummitAttendeeNote
+     * @throws EntityNotFoundException
+     */
+    public static function populate
+    (
+        SummitAttendee      $attendee,
+        SummitAttendeeNote  $attendee_note,
+        array               $payload
+    )
+    {
+        $ticket = null;
+        $author = null;
+
+        if (isset($payload['ticket_id'])) {
+            $ticket_id = intval($payload['ticket_id']);
+            $ticket = $attendee->getTicketById($ticket_id);
+            if (is_null($ticket))
+                throw new EntityNotFoundException(sprintf("Ticket id %s does not belong to attendee id %s.",
+                    $ticket_id, $attendee->getId()));
+        }
+
+        if (isset($payload['author_id'])) {
+            $member_repository = EntityManager::getRepository(Member::class);
+            $author = $member_repository->getById(intval($payload['author_id']));
+            if (!$author instanceof Member)
+                throw new EntityNotFoundException("Member not found.");
+        }
+
+        $attendee_note->setAuthor($author);
+        $attendee_note->setTicket($ticket);
```

Comment:
this could potentially set the ticket id to null this should be enclosed on 
   if (isset($payload['ticket_id'])) {